<launch>
	<arg name="serial_number" default=""/> <!-- specify serial number of your camera -->
	<arg name="color_width" default="640"/>
	<arg name="color_height" default="360"/>
	<arg name="depth_width" default="640"/>
	<arg name="depth_height" default="480"/>
	<arg name="publish_image" default="false"/>
	<arg name="camera_frame" default="b_bot_camera"/>
	<arg name="camera_data_path" default="$(find o2as_realsense_camera)/image"/>
	<arg name="color_image_filename" default="color.png"/>
	<arg name="depth_image_filename" default="depth.png"/>
	<arg name="point_cloud_filename" default="cloud.pcd"/>
	
	<arg name="node_name" default="camera"/>
	<node name="$(arg node_name)" pkg="o2as_realsense_camera" type="o2as_realsense_camera_node" output="screen">
		<param name="serial_number" type="string" value="$(arg serial_number)"/>
		<param name="color_width" type="int" value="$(arg color_width)"/>
		<param name="color_height" type="int" value="$(arg color_height)"/>
		<param name="depth_width" type="int" value="$(arg depth_width)"/>
		<param name="depth_height" type="int" value="$(arg depth_height)"/>
		<param name="publish_image" type="bool" value="$(arg publish_image)"/>
		<param name="camera_frame" type="string" value="$(arg camera_frame)"/>
		<param name="camera_data_path" type="string" value="$(arg camera_data_path)"/>
		<param name="color_image_filename" type="string" value="$(arg color_image_filename)"/>
		<param name="depth_image_filename" type="string" value="$(arg depth_image_filename)"/>
		<param name="point_cloud_filename" type="string" value="$(arg point_cloud_filename)"/>
	</node>
</launch>
